OpenRA.Mods.Common/UpdateRules/Rules/20250330/ReplaceBaseAttackNotitifer.cs

```diff
@@ -0,0 +1,75 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	/// <summary>
+	/// Replaces the BaseAttackNotifier with a new AttackNotifier that uses the
+	/// new attack system.
+	/// </summary>
+	public class ReplaceBaseAttackNotifier : UpdateRule, IBeforeUpdateActors
+	{
+		public override string Name => "Replace BaseAttackNotifier with DamageNotifier";
+		public override string Description => "Replaces the BaseAttackNotifier with a new DamageNotifier that uses BaseBuilding target type.";
+
+		bool any = false;
+		readonly Dictionary<string, string> notifications = [];
+
+		IEnumerable<string> IBeforeUpdateActors.BeforeUpdateActors(ModData modData, List<MiniYamlNodeBuilder> resolvedActors)
+		{
+			any = false;
+			foreach (var actor in resolvedActors)
+			{
+				foreach (var notifier in actor.ChildrenMatching("BaseAttackNotifier"))
+				{
+					var notification = notifier.LastChildMatching("Notification", false);
+					if (notification != null)
+						notifications.Add(actor.Key + ":" + notifier.Key, notification.Value.Value);
+				}
+			}
+
+			yield break;
+		}
+
+		public override IEnumerable<string> UpdateActorNode(ModData modData, MiniYamlNodeBuilder actorNode)
+		{
+			foreach (var notifier in actorNode.ChildrenMatching("BaseAttackNotifier"))
+			{
+				any = true;
+				notifier.RenameKey("DamageNotifier");
+				notifier.AddNode("ValidTargets", "BaseBuilding");
+				if (notifications.TryGetValue(actorNode.Key + ":" + notifier.Key, out var notificationValue))
+				{
+					var notification = notifier.LastChildMatching("Notification", false);
+					if (notification != null)
+						notification.Value.Value = notificationValue;
+					else
+						notifier.AddNode("Notification", notificationValue);
```

Comment:
if you found the value, shouldn't you just do nothing? The value will be correctly inherited from parents